### ISSUE TITLE
Strategy for preparing to have more warnings from all.sh

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#7894]
When we make changes to `all.sh` that result in more compiler warnings, there's a risk that pull requests from branches forked before the changes pass the CI (which only tests the pull request itself) but the result of merging them won't pass the CI. At the moment, we don't run the CI on merges, so we'll know something breaks due to the nightly.

We are going to make such a change soon: https://github.com/Mbed-TLS/mbedtls/issues/7891, likely with more to come in the next few months.

The goal of this issue is to either decide that we don't mind, or find a strategy to avoid this problem. We may create a follow-up issue to implement the chosen strategy. Some potential strategies:

* [Use GitHub merge queues](https://github.com/Mbed-TLS/mbedtls-test/issues/105). Are we ready to commit to switching them on? Should we try them out and how?
* Turn the full pr-merge back on. Downside: significantly increases the CI load.
* Replace pr-head with pr-merge. Downside: reduces the usefulness of the CI.
* Run a test campaign on the merges of open pull requests, but only once. How?
* Add a reduced pr-merge job that only tests the changed part during a transition period.
* Add a Travis job corresponding to the changed part during a transition period.

**Comment by gilles-peskine-arm** [comment]
We've discussing this internally, and we've settled on enabling merge queues. Actions:

1. Bence is working on enabling merge queues.
2. Bence and I will ensure that the checks in the merge queue includes at least the high-risk items at any given time, e.g. new all.sh jobs that risk emitting more warnings.

**Comment by mpg** [comment]
We are using the merge queue and it runs the full `all.sh` AFAIK, so I'm guessing this issue can be closed now?

**Comment by gilles-peskine-arm** [comment]
Goal fulfilled by using merge queues.
